File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/WeaponToUI.java
/**
 * @file WeaponToUI.java
 * @author Profesor
 */
package deepspace;

/**
 * @brief Representación ToUI de Weapon
 */
public class WeaponToUI {
    private WeaponType type;
    private float power;
    private int uses;  
    
    /**
     * @brief Constructor con un parámetro
     * @param w Instancia de la clase Weapon
     */
    WeaponToUI(Weapon w) {
        type=w.getType();
        power=w.power();
        uses=w.getUses();
    }  

    /**
     * @brief Consultor del tipo de arma
     * @return type
     */
    public WeaponType getType() {
        return type;
    }

    /**
     * @brief Consultor del poder del arma
     * @return power
     */
    public float getPower() {
        return power;
    }

    /**
     * @brief Consultor de los usos
     * @return uses
     */
    public int getUses() {
        return uses;
    }
}

File: practicas/practica_2/Deepspace_Ruby_P2/lib/WeaponToUI.rb
#encoding:utf-8

# File: WeaponToUI.rb
# Author: Profesor

module Deepspace;

# Brief: Representación ToUI de Weapon
class WeaponToUI 

  # MIGUEL: por que  name  no forma parte de esta clase ?
  
    # Consultores
    attr_reader :type, :power, :uses
    
    # Brief: Constructor con un parámetro
    # Param w: Instancia de la clase Weapon
    def initialize(w) 
        @type=w.type();
        @power=w.power();
        @uses=w.uses();
    end  
    
      # Brief: Función para representar una instancia de la clase WeaponToUI
  #        en un string
  # Return: out String que representa una instancia de la clase WeaponToUI
    def to_s
      return "Type: #{@type}, Power: #{@power}, Uses: #{@uses}"
    end
    
end # Class

end # Module